src/os/bluestore/BlueStore.cc

```diff
@@ -3610,7 +3612,11 @@ void BlueStore::ExtentMap::reshard(
   }
 
   fault_range(db, needs_reshard_begin, (needs_reshard_end - needs_reshard_begin));
-
+  if (needs_reshard_end == OBJECT_MAX_SIZE) {
+    data_reshard_end = extent_map.empty() ? needs_reshard_end : extent_map.rbegin()->blob_end();
```

Comment:
= -> ==

Reply:
Blush.

Reply:
how about:
`data_reshard_end = (extent_map.empty() ? needs_reshard_end : extent_map.rbegin()->blob_end());`
?
Because it seems priority of = and ? are confusing here, but "right to left" evaluation makes it work OK.